pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -0,0 +1,216 @@
+		}
+		var healthzServer *healthcheck.HealthzServer
+		if len(c.ProxyConfig.HealthzBindAddress) > 0 {
+			healthzServer = healthcheck.NewDefaultHealthzServer(c.ProxyConfig.HealthzBindAddress, 2*c.ProxyConfig.IPTables.SyncPeriod.Duration)
+		}
+		if c.ProxyConfig.IPTables.MasqueradeBit == nil {
+			// IPTablesMasqueradeBit must be specified or defaulted.
+			glog.Fatalf("Unable to read IPTablesMasqueradeBit from config")
+		}
+		proxierIptables, err := iptables.NewProxier(
+			iptInterface,
+			utilsysctl.New(),
+			execer,
+			c.ProxyConfig.IPTables.SyncPeriod.Duration,
+			c.ProxyConfig.IPTables.MinSyncPeriod.Duration,
+			c.ProxyConfig.IPTables.MasqueradeAll,
+			int(*c.ProxyConfig.IPTables.MasqueradeBit),
+			c.ProxyConfig.ClusterCIDR,
+			hostname,
+			bindAddr,
+			recorder,
+			healthzServer,
+		)
+
+		if err != nil {
+			glog.Fatalf("error: Could not initialize Kubernetes Proxy. You must run this process as root (and if containerized, in the host network namespace as privileged) to use the service proxy: %v", err)
+		}
+		proxier = proxierIptables
+		endpointsHandler = proxierIptables
+		servicesHandler = proxierIptables
+		// No turning back. Remove artifacts that might still exist from the userspace Proxier.
+		glog.V(0).Info("Tearing down userspace rules.")
+		userspace.CleanupLeftovers(iptInterface)
+	case componentconfig.ProxyModeUserspace:
+		glog.V(0).Info("Using userspace Proxier.")
+		// This is a proxy.LoadBalancer which NewProxier needs but has methods we don't need for
+		// our config.EndpointsHandler.
+		loadBalancer := userspace.NewLoadBalancerRR()
+		// set EndpointsHandler to our loadBalancer
+		endpointsHandler = loadBalancer
+
+		execer := utilexec.New()
+		proxierUserspace, err := userspace.NewProxier(
+			loadBalancer,
+			bindAddr,
+			iptInterface,
+			execer,
+			*portRange,
+			c.ProxyConfig.IPTables.SyncPeriod.Duration,
+			c.ProxyConfig.IPTables.MinSyncPeriod.Duration,
+			c.ProxyConfig.UDPIdleTimeout.Duration,
+		)
+		if err != nil {
+			glog.Fatalf("error: Could not initialize Kubernetes Proxy. You must run this process as root (and if containerized, in the host network namespace as privileged) to use the service proxy: %v", err)
+		}
+		proxier = proxierUserspace
+		servicesHandler = proxierUserspace
+		// Remove artifacts from the pure-iptables Proxier.
+		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
+		iptables.CleanupLeftovers(iptInterface)
+	default:
+		glog.Fatalf("Unknown proxy mode %q", c.ProxyConfig.Mode)
+	}
+
+	// Create configs (i.e. Watches for Services and Endpoints)
+	// Note: RegisterHandler() calls need to happen before creation of Sources because sources
+	// only notify on changes, and the initial update (on process start) may be lost if no handlers
+	// are registered yet.
+	serviceConfig := pconfig.NewServiceConfig(
+		c.InternalKubeInformers.Core().InternalVersion().Services(),
+		c.ProxyConfig.ConfigSyncPeriod.Duration,
+	)
+
+	if c.EnableUnidling {
```

Comment:
There are definitely follow-up questions about the unidler, but this probably isn't making it worse.

Side question, where does the idler fall in the update order with controllers.  Will we have to have a version where we have a controller running on its own resource and events?